### ISSUE TITLE
warn for `dbt_project.yml` configs which do not affect any models

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#691]
This happens _far_ too frequently and dbt offers no help. It typically occurs because users do not include the project name in their `models:` dict, eg:

```
models:
  project_name:
    first_path:
      materialized: ephemeral
```

This is particularly confusing because 1) the `models/` path is implicit in the hierarchical config, 2) the project name is required, but is not a part of the path to the file, 3) our docs here are probably insufficient.

dbt should handle these configurations in a better way, ideally without breaking backwards compatibility. A good first step could be to identify when configurations do not affect any models. It could also include a helpful hint if config paths are valid, but missing a project name.

**Comment by drewbanin** [comment]
Fixed in #725

**Comment by kapoorsaumya08** [comment]
hi i am getting the same error despite having my model in place. stuck on this since 2 days.
